docs/tutorials/tutorial.e2e-ssl.md

```diff
@@ -0,0 +1,200 @@
+```
+
+Verify that you can curl the application
+```bash
+> kubectl get pods
+NAME                                 READY   STATUS    RESTARTS   AGE
+website-deployment-9c8c6df7f-5bqwh   1/1     Running   0          24s
+website-deployment-9c8c6df7f-wxtnp   1/1     Running   0          24s
+
+> kubectl exec -it website-deployment-9c8c6df7f-5bqwh -- curl -k https://localhost:8443
+Hello World!
+```
+
+## Upload the backend certificate's root certificate to Application Gateway
+When you are setting up SSL between Application Gateway and Backend, if you are using a self-signed certificate or a certificate signed by a custom root CA on the backend, then you need to upload self-signed or the Custom root CA of the backend certificate on the Application Gateway.
+
+```bash
+applicationGatewayName="<gateway-name>"
+resourceGroup="<resource-group>"
+az network application-gateway root-cert create \
+    --gateway-name $applicationGatewayName  \
+    --resource-group $resourceGroup \
+    --name backend-tls \
+    --cert-file backend.crt
+```
+
+## Setup ingress for E2E
+Now, we will configure our ingress to use the `frontend` certificate for frontend SSL and `backend` certificate as root certificate so that Application Gateway can authenticate the backend.
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: website-ingress
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+    appgw.ingress.kubernetes.io/ssl-redirect: "true"
+    appgw.ingress.kubernetes.io/backend-protocol: "https"
+    appgw.ingress.kubernetes.io/backend-hostname: "backend"
+    appgw.ingress.kubernetes.io/appgw-trusted-root-certificate: "backend-tls"
+spec:
+  tls:
+    - secretName: frontend-tls
+      hosts:
+        - website.com
+  rules:
+    - host: website.com
+      http:
+        paths:
+        - path: /
+          backend:
+            serviceName: website-service
+            servicePort: 8443
+EOF
+```
+
+For frontend SSL, we have added `tls` section in our ingress resource.
+```yaml
+  tls:
+    - secretName: frontend-tls
+      hosts:
+        - website.com
+```
+
+For backend SSL, we have added the following annotations:
+```yaml
+appgw.ingress.kubernetes.io/backend-protocol: "https"
+appgw.ingress.kubernetes.io/backend-hostname: "backend"
+appgw.ingress.kubernetes.io/appgw-trusted-root-certificate: "backend-tls"
+```
+
+Here, it is important to note that `backend-hostname` should be the hostname that the backend will accept and it should also match with the Subject/Subject Alternate Name of the certificate used on the backend.
+
+
+After you have successfully completed all the above steps, you should be able to see the ingress's IP address and visit the website.
+```
+> kubectl get ingress
+NAME              HOSTS         ADDRESS         PORTS     AGE
+website-ingress   website.com   <gateway-ip>   80, 443   36m
+
+> curl -k -H "Host: website.com" https://<gateway-ip>
```

Comment:
doesn't match to the port exposed in ingress above

Reply:
Again, 8443 is the port used on the service.
On frontend, we only support 80 and 443.